Source/JavaScriptCore/CMakeLists.txt

```diff
@@ -159,6 +159,7 @@ add_custom_command(
     OUTPUT ${JavaScriptCore_DERIVED_SOURCES_DIR}/JSCBytecodeCacheVersion.cpp
     MAIN_DEPENDENCY ${JAVASCRIPTCORE_DIR}/runtime/JSCBytecodeCacheVersion.cpp.in
     COMMAND ${PERL_EXECUTABLE} -pe s/CACHED_TYPES_CKSUM/__TIMESTAMP__/ ${JAVASCRIPTCORE_DIR}/runtime/JSCBytecodeCacheVersion.cpp.in > ${JavaScriptCore_DERIVED_SOURCES_DIR}/JSCBytecodeCacheVersion.cpp
+    COMMAND touch -r ${JAVASCRIPTCORE_DIR}/runtime/JSCBytecodeCacheVersion.cpp.in ${JavaScriptCore_DERIVED_SOURCES_DIR}/JSCBytecodeCacheVersion.cpp
```

Comment:
This is actually a pretty good idea!

Except you can't use `touch` here, because it doesn't exist on Windows and this is a cross-platform file (which is why the Windows CI failed). I looked for an equivalent CMake command and found [this](https://cmake.org/cmake/help/latest/command/file.html#touch), but it's not good enough for what you're trying to do.

We'll have to think of something else.

Reply:
I'm not sure whether you could support it in all the environments you need to build in, but one thing we considered in https://github.com/WebKit/WebKit/commit/adb96155be5e75707c6d831c3329b72b0afcc2b1 was using a string from the git repo's state (e.g. `git describe --dirty`).

Reply:
Hang on, can we just use [`cmake -E touch`](https://cmake.org/cmake/help/latest/manual/cmake.1.html#cmdoption-cmake-E-arg-touch)?

Reply:
Another option: if there's a way to test whether `touch` is installed, then you we could have reproducible builds on systems that do have touch.

Reply:
> Another option: if there's a way to test whether `touch` is installed, then we could have reproducible builds on systems that do have `touch`.

I've adjusted the commit to do that, please review.